feat(Form): expose loading state to default slot

dan-hale opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue

<!-- If it resolves an open issue, please link the issue here. For example "Resolves #123" -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Loading state for onSubmit should be exposed to form slot to enable displaying loading states while submission is pending.
<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

dan-hale committed:
feat(Form): Expose loading state to slot

benjamincanac commented on src/runtime/components/Form.vue:
```diff
@@ -315,6 +315,6 @@ defineExpose<Form<I>>({
     @submit.prevent="onSubmitWrapper"
   >
-    <slot :errors="errors" />
+    <slot :errors="errors" :loading />
```
```suggestion
    <slot :errors="errors" :loading="loading" />
```

dan-hale replied:
Just removed it. I Had accidentally left this in while I was considering emitting an event for the result of onSubmit.

benjamincanac commented on src/runtime/components/Form.vue:
```diff
@@ -240,7 +240,7 @@ async function onSubmitWrapper(payload: Event) {
   try {
     event.data = await _validate({ nested: true, transform: props.transform })
-    await props.onSubmit?.(event)
+    const result = await props.onSubmit?.(event)
```
What's this for?

dan-hale committed:
fix(Form): Remove mistake

Accidentally left this in, was considering emitting an event for the result of onSubmit.

dan-hale committed:
fix(Form): Semantics

Co-authored-by: Benjamin Canac <canacb1@gmail.com>
